### DISCUSSION TITLE
Select all between nearest selected and current focused file

### DISCUSSION THREAD

**Discussion by perdakovich** [gokcehan/lf#1631] [Ideas]
In graphical file managers I often use the Shift Select function, when you hold Shift and click on a file, it selects all the files between the current selection and the clicked file. 
In **lf**, I often think "Damn, it would be nice to use shift select here..." to select a bunch of files instead of tediously holding down the spacebar...

I believe that something similar should be implemented in **lf** as well, it would greatly improve selection process.

@DusanLesan talked about this idea here, so you should read it:
https://github.com/gokcehan/lf/pull/1101#issuecomment-1416714201

**Comment by DusanLesan** [comment]
> @DusanLesan talked about this idea here, so you should read it: [#1101 (comment)](https://github.com/gokcehan/lf/pull/1101#issuecomment-1416714201)

I tend to agree with Ilya about it being bad to need to count in your head so you know which selection is closer for shift select. I prefer the idea of always selecting from above if there is more than one selection. For example, if the selection above is 10 rows before the current one, I would still prefer it over any selection after, even if the current one is only 2 files below. Another idea is to select everything between the current row and the last one that was selected, but that could be difficult since we may need to remember the selection time for the last selection in each directory. Maybe even better would be to completely detach the starting point from the selection and have the user select the start point as a different type of file state. However, these are all just ideas since I have gotten used to invert-below and it fulfills all my needs.

**Reply by perdakovich** [reply]
Honestly I don't really understand how to use invert-below as a replacement for shift select.
Well, it inverts below the cursor, so what? I'd be glad if you could explain the concept to me

**Reply by DusanLesan** [reply]
`invert-below` allows you to select files in some range without knowing exact number of files you want to select. So you would invert all below the starting point, move to the end point and invert again returning all except ones between start and end point as they were before

https://github.com/gokcehan/lf/assets/20520590/a099699e-8512-482a-912d-c8df25f7ea68

**Reply by perdakovich** [reply]
Yeah, now I see, thanks!
What key did you bind it on?
Also what recording program do you have, I see its logo in the corner.

**Reply by DusanLesan** [reply]
I have used `V` as a key. I think it was initially suggested as a default since `v` toggles all.
The icon in tray is for slack. Not related to recordings. I use `ffmpeg` command from a [script](https://github.com/DusanLesan/dotfiles/blob/master/.local/bin/record) to do recording

**Reply by perdakovich** [reply]
Ah I thought it was a recording software because of the red circle.
Yeah `V` is a nice bind  :+1:
